nym-node-status-api/nym-node-status-api/src/monitor/mod.rs

```diff
@@ -70,345 +85,342 @@ pub(crate) async fn spawn_in_background(
-    tracing::debug!("Of those, {} mixnodes", mixnodes.len());
+        let gateways = all_nodes
+            .iter()
+            .filter(|node| node.description.declared_role.entry)
+            .collect::<Vec<_>>();
+        tracing::debug!("Of those, {} gateways", gateways.len());
 
-    log_gw_in_explorer_not_api(explorer_gateways.as_slice(), gateways.as_slice());
+        let mixnodes = all_nodes
+            .iter()
+            .filter(|node| node.description.declared_role.mixnode)
+            .collect::<Vec<_>>();
+        tracing::debug!("Of those, {} mixnodes", mixnodes.len());
+
+        let bonded_node_info = api_client
+            .get_all_bonded_nym_nodes()
+            .await?
+            .into_iter()
+            .map(|node| (node.bond_information.node_id, node.bond_information))
+            // for faster reads
+            .collect::<HashMap<_, _>>();
+
+        let mut gateway_geodata = Vec::new();
+        for gateway in gateways.iter() {
+            if let Some(node_info) = bonded_node_info.get(&gateway.node_id) {
+                let gw_geodata = GatewayGeoData {
+                    identity_key: node_info.node.identity_key.to_owned(),
+                    owner: node_info.owner.to_owned(),
+                    pledge_amount: node_info.original_pledge.to_owned(),
+                    location: self.location_cached(&gateway).await,
+                };
+                gateway_geodata.push(gw_geodata);
+            }
+        }
 
-    let all_skimmed_nodes = api_client
-        .get_all_basic_nodes(None)
-        .await
-        .log_error("get_all_basic_nodes")?;
+        // contains performance data
+        let all_skimmed_nodes = api_client
+            .get_all_basic_nodes(None)
+            .await
+            .log_error("get_all_basic_nodes")?;
 
-    let mixnodes = api_client
-        .get_cached_mixnodes()
-        .await
-        .log_error("get_cached_mixnodes")?;
-    tracing::debug!("Fetched {} mixnodes", mixnodes.len());
+        let gateways_blacklisted = all_skimmed_nodes
+            .iter()
+            .filter_map(|node| {
+                if node.performance.round_to_integer() <= 50 && node.supported_roles.entry {
+                    Some(node.ed25519_identity_pubkey.to_base58_string())
+                } else {
+                    None
+                }
+            })
+            .collect::<HashSet<_>>();
+
+        // Cached mixnodes don't include blacklisted nodes
+        // We need that to calculate the total locked tokens later
+        // TODO dz deprecated API, remove
+        let mixnodes = api_client
+            .nym_api
+            .get_mixnodes_detailed_unfiltered()
+            .await
+            .log_error("get_mixnodes_detailed_unfiltered")?;
+        let mixnodes_described = api_client
+            .nym_api
+            .get_mixnodes_described()
+            .await
+            .log_error("get_mixnodes_described")?;
+        let mixnodes_active = api_client
```

Comment:
I'm not sure this is an endpoint you want to be using. it will only give you **old legacy** mixnodes that happen to be in the rewarded set. I reckon you might get maybe like 10-20 of them

Reply:
noted for upcoming refactor